doc_source/sam-resource-function.md

```diff
@@ -265,23 +265,38 @@ The following is an example of an AWS::Serverless::Function that uses InlineCode
 ```
 Type: AWS::Serverless::Function
 Properties:
+  VersionDescription: Initial Commit
+  AutoPublishAlias: staging
   Handler: index.handler
   Runtime: python3.6
   InlineCode: |
     def handler(event, context):
       print("Hello, world!")
   ReservedConcurrentExecutions: 30
+  Environment:
+    Variables:
+      TABLE_NAME: my-table
   Layers:
-    - Ref: MyLayer
+    - !Sub arn:${AWS::Partition}:lambda:${AWS::Region}:923456789012:layer:MyLayer:1
   Tracing: Active
   Timeout: 120
   Policies:
-    - AWSLambdaExecute
+    - DynamoDBCrudPolicy:
+        TableName: !Sub my-table
     - Version: '2012-10-17' 
       Statement:
         - Effect: Allow
           Action:
             - s3:GetObject
             - s3:GetObjectACL
           Resource: 'arn:aws:s3:::my-bucket/*'
-```
\ No newline at end of file
+  Events:
+    PhotoUpload:
+      Type: S3
+      Properties:
+        Bucket: my-photo-bucket # bucket must be created in the same template
+  Tags:
+    AppNameTag: ThumbnailApp
+    DepartmentNameTag: ThumbnailDepartment
```

Comment:
Most important edit in the YAML example. The rest were properties I've found important in this context. I apologize for not making a separate PR as I now realize I should have.